packages/lesswrong/server/authentication/auth0.ts

```diff
@@ -0,0 +1,55 @@
+import { DatabaseServerSetting } from "../databaseSettings";
+import { ManagementClient } from "auth0";
+import Profile from "passport-auth0/lib/Profile";
+
+type Auth0Settings = {
+  appId: string;
+  secret: string;
+  domain: string;
+  originalDomain: string;
+}
+
+/**
+ * Auth0 Management API Client
+ * For this to work, we must authorize the forum application in Auth0:
+ * Applications -> APIs -> Auth0 Management API -> Machine to Machine Applications
+ */
+const auth0Client = new class Auth0Client {
+  private settings = new DatabaseServerSetting<Auth0Settings|null>("oAuth.auth0", null);
+  private client?: ManagementClient;
+
+  get() {
+    if (!this.client) {
+      const {
+        appId: clientId,
+        secret: clientSecret,
+        originalDomain: domain,
+      } = this.settings?.get() ?? {};
+      if (!clientId || !clientSecret || !domain) {
+        throw new Error("Missing auth0 environment settings");
+      }
+      const scope = "read:users update:users";
+      this.client = new ManagementClient({clientId, clientSecret, domain, scope});
+    }
+
+    return this.client;
+  }
+}
+
+const getAuth0Id = (user: DbUser) => {
+  const id = user.services?.auth0?.id ?? user.services?.auth0?.user_id;
+  if (!id) {
+    throw new Error("User does not have an Auth0 user ID");
+  }
+  return id;
+}
```

Comment:
In this case, I think we should disable the email address field in the form, instead of throwing an error after the form is submitted. I'm guessing a similar issue is happening when I try to change the email address of my account created via oauth, since I see "Cannot update email for this user" when I submit the form.